src/main/webapp/app/pool/pool-list/pool-list.service.ts

```diff
@@ -3,13 +3,56 @@ import { Observable } from 'rxjs';
 import { Pool } from 'app/pool/pool';
 import { HttpClient, HttpResponse } from '@angular/common/http';
 import { SERVER_API_URL } from 'app/app.constants';
+import { PoolListComponent } from 'app/pool/pool-list/pool-list.component';
+import { JhiAlertService } from 'ng-jhipster';
+import { Tool } from 'app/pool/tool';
+
 @Injectable({
     providedIn: 'root'
 })
 export class PoolListService {
-    constructor(private http: HttpClient) {}
+    private link: string;
+    alerts: any[];
+
+    constructor(private http: HttpClient, private alertService: JhiAlertService) {
+        this.alerts = this.alertService.get();
+    }
 
     getPools(): Observable<HttpResponse<Pool[]>> {
         return this.http.get<Pool[]>(SERVER_API_URL + 'api/pool/list', { observe: 'response' });
     }
+
+    getTools(): Observable<HttpResponse<Tool[]>> {
+        return this.http.get<Tool[]>(SERVER_API_URL + 'api/tool/list', { observe: 'response' });
+    }
+
+    getFilteredPools(poolListComponent: PoolListComponent): Observable<HttpResponse<Pool[]>> {
+        this.link = SERVER_API_URL + 'api/pool/list-filter/params?params=';
```

Comment:
It just look bad, it should be 'api/pool/list-filter?search=' if you change param name and path on server side as I suggested before.